lib/interactor/contracts/dsl.rb

```diff
@@ -73,6 +73,7 @@ def contract
       def expects(&block)
         contract.add_expectation(&block)
         define_expectations_hook
+        def_delegates_to_context(contract.expectations.keys)
```

Comment:
Since this is defined in another module, it is hard to reason about where this came from. Did you consider just keeping the entire DSL together instead of creating the `Forwarder` module?

I like the idea that the forwarding work is self-contained, but I'm not a fan of this leaking out.

What if we implemented the forwarding via an override of `expects` in `Forwarder` (that calls `super`)? That would be the best of both worlds, I think ... assuming it works.